From ef89004c517e6498f0005f007a4b52659dd72a0a Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 23 May 2019 15:42:45 +0800
Subject: [PATCH] snmp: fix avc denials for snmpd_t

Fix:
type=AVC msg=audit(1558592078.824:237): avc:  denied  { map } for
pid=239 comm="snmpd" path="/etc/passwd" dev="vda" ino=31098
scontext=system_u:system_r:snmpd_t tcontext=system_u:object_r:etc_t
tclass=file permissive=0

type=AVC msg=audit(1558592078.846:241): avc:  denied  { read } for
pid=239 comm="snmpd" name="hwdb.bin" dev="vda" ino=27684
scontext=system_u:system_r:snmpd_t
tcontext=system_u:object_r:systemd_hwdb_t tclass=file permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/services/snmp.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/services/snmp.te b/policy/modules/services/snmp.te
index f778629..1fc027f 100644
--- a/policy/modules/services/snmp.te
+++ b/policy/modules/services/snmp.te
@@ -94,8 +94,12 @@ domain_signull_all_domains(snmpd_t)
 domain_read_all_domains_state(snmpd_t)
 domain_exec_all_entry_files(snmpd_t)
 
+systemd_map_hwdb(snmpd_t)
+systemd_read_hwdb(snmpd_t)
+
 files_read_usr_files(snmpd_t)
 files_read_etc_runtime_files(snmpd_t)
+files_map_etc_files(snmpd_t)
 files_search_home(snmpd_t)
 
 fs_getattr_all_dirs(snmpd_t)
-- 
2.7.4

